ibis/backends/clickhouse/datatypes.py

```diff
@@ -151,6 +152,44 @@ def struct():
             nullable=False,
         )
 
+    @p.generate
+    def enum_value():
+        yield dt.SPACES
+        key = yield dt.RAW_STRING
+        yield dt.spaceless_string('=')
+        value = yield p.digit.at_least(1).concat()
+        return (key, int(value))
+
+    @p.generate
+    def lowcardinality():
+        yield dt.spaceless_string('LowCardinality')
+        yield dt.LPAREN
+        r = yield ty
+        yield dt.RPAREN
+        return r
+
+    @p.generate
+    def enum():
+        yield dt.spaceless_string('enum')
+        enumsz = yield p.digit.at_least(1).concat()
+        enumsz = int(enumsz)
+        yield dt.LPAREN
+        yield enum_value.sep_by(dt.COMMA).map(dict)  # ignore values
+        yield dt.RPAREN
+        if enumsz == 8:
+            return dt.Int8(nullable=False)
+        else:
+            return dt.Int64(nullable=False)
+
+    @p.generate
+    def other():
+        r = yield p.regex(r'.*')
+        warnings.warn(f'Unknown Clickhouse type "{r}"')
+        return dt.String(nullable=True)
```

Comment:
I think this has limited utility and will make whatever columns are inferred here unusable in any subsequent queries.

Ibis may happily plod along with a string type, but the code it generates is extremely unlikely to work once you query the database.

Types in ibis determine the API, so whatever expression is a `String`-but-not-really will have APIs like `lpad` and `substr` which likely do not make any sense for the type that couldn't be inferred.

ClickHouse does no implicit casting, and since the user is presented with APIs that aren't valid there's not much reason to "keep going" only to fail later.

Reply:
Yeah, I came across this with the Enum type too, which doesn't work for even simple comparisons (i8 != string).  I'll remove and let it fail as before.